### PR TITLE
Issue #420: Create Separate Modules

### PR DESCRIPTION
**Pull request by JosepSampe**
## Description

This PR creates different modules for the core, delta and api code

## Type of change

Refactoring

## Checklist:

Here is the list of things you should do before submitting this pull request:

- [x] New feature / bug fix has been committed following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add logging to the code following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add comments to the code (make it easier for the community!).
- [x] Change the documentation.
- [x] Add tests.
- [x] Your branch is updated to the main branch (dependent changes have been merged).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -280,8 +280,8 @@ To publish a new version of the qbeast-spark project, follow these steps:
 export QBEAST_SPARK_VERSION=0.6.0-SNAPSHOT
 $SPARK_350/bin/spark-shell --repositories https://s01.oss.sonatype.org/content/repositories/snapshots \
 --packages io.delta:delta-spark_2.12:3.1.0,io.qbeast:qbeast-spark_2.12:$QBEAST_SPARK_VERSION \
---conf spark.sql.extensions=io.qbeast.spark.delta.QbeastDeltaSparkSessionExtension \
---conf spark.sql.catalog.spark_catalog=io.qbeast.spark.internal.sources.catalog.QbeastCatalog
+--conf spark.sql.extensions=io.qbeast.sql.QbeastSparkSessionExtension \
```
Comment (osopardo1) [inline CONTRIBUTING.md:283]: Mmm, I am not entirely sure of removing the "spark" name on `io.qbeast.spark` package. Even if we don't have a module named "spark" in the `sbt` structure, I think it should be explicit that all this interfaces are for the spark `implementation`.

**Review by osopardo1 — APPROVED** [review]
Just a comment regarding package names
 Otherwise lgtm

File: build.sbt [hunk 2]
```diff
@@ -33,7 +29,9 @@ lazy val qbeastSpark = (project in file("."))
     name := "qbeast-spark",
     libraryDependencies ++= Seq(
       sparkSql % Provided,
-      deltaSpark % Provided),
+      deltaSpark % Provided,
+      sparkml % Test,
```
Comment (cugni) [inline build.sbt:33]: why do we need sparkml in test?

File: src/main/scala/io/qbeast/table/IndexedTable.scala [hunk 3]
```diff
@@ -459,6 +461,12 @@ private[table] class IndexedTableImpl(
     stagingDataManager.updateWithStagedData(data) match {
       case r: StagingResolution if r.sendToStaging =>
         stagingDataManager.stageData(data, indexStatus, options, append)
+        if (snapshot.isInitial) {
```
Comment (cugni) [inline src/main/scala/io/qbeast/table/IndexedTable.scala:464]: Is this tested?

File: core/src/main/scala/org/apache/spark/sql/SchemaUtils.scala [hunk 4]
```diff
@@ -85,11 +85,11 @@ object SchemaUtils {
       val existingFieldNames = existing.fieldNames.map(_.toLowerCase).toSet
       assert(
         existingFieldNames.size == existing.length,
-        "Delta tables don't allow field names that only differ by case")
+        "Delta table don't allow field names that only differ by case")
```
Comment (cugni) [inline core/src/main/scala/org/apache/spark/sql/SchemaUtils.scala:88]: keep the plural here "Delta tables"

**Review by cugni — APPROVED** [review]

**Pull request merged**